### PR TITLE
Feat: added account suffix

### PR DESCRIPTION
**Pull request by victoraugustolls**

### CODE REVIEW

File: app/domain/vos/account_name.go [hunk 1]
```diff
@@ -54,18 +61,31 @@ func NewAccountName(name string) (*AccountName, error) {
...
+		suffix = identifiers[1]
+	}
+
 	return &AccountName{
 		Class:    accountClass,
 		Group:    levels[groupLevel],
 		Subgroup: levels[subgroupLevel],
-		ID:       levels[idLevel],
+		ID:       identifiers[0],
+		Suffix:   suffix,
```
Comment (victoraugustolls) [inline app/domain/vos/account_name.go:77]: I didn't want to use fixed indexes here (like `identifiers[0]`),  and I also think that the name `identifiers` is a bad one. Any suggestions?

File: app/domain/vos/account_name.go [hunk 2]
```diff
@@ -23,6 +24,11 @@ const (
 //   - group, subgroup and id are "free text";
 //   - ":" must be used as a separator.
 //
+// AccountName can have a fifth level in her structure: "class:group:subgroup:id/suffix", where:
```
Comment (Streppel) [inline app/domain/vos/account_name.go:27]:
```suggestion
// AccountName can have a fifth level in its structure: "class:group:subgroup:id/suffix", where:
```

File: app/domain/vos/account_name.go [hunk 3]
```diff
@@ -54,18 +61,31 @@ func NewAccountName(name string) (*AccountName, error) {
 		return nil, app.ErrInvalidAccountStructure
 	}
 
+	var suffix string
+	identifiers := strings.SplitN(levels[idLevel], AccountSuffixSep, 2)
+	if len(identifiers) == 1 {
+		suffix = ""
+	} else {
+		suffix = identifiers[1]
+	}
```
Comment (Streppel) [inline app/domain/vos/account_name.go:70]:
```suggestion
	if len(identifiers) > 1 {
		suffix = identifiers[1]
	}
```

Comment (Streppel) [reply app/domain/vos/account_name.go:70]: What do you think of this? Getting rid of an unecessary else

File: app/domain/vos/account_name.go (hunk 1 above)
Comment (Streppel) [reply app/domain/vos/account_name.go:77]: identifiers can hold both id and probably suffix right?

Comment (victoraugustolls) [reply app/domain/vos/account_name.go:77]: Yes!

File: app/domain/vos/account_name.go (hunk 2 above)
Comment (victoraugustolls) [reply app/domain/vos/account_name.go:27]: Changed both, old and new docs!

File: app/domain/vos/account_name.go (hunk 3 above)
Comment (victoraugustolls) [reply app/domain/vos/account_name.go:70]: Way better! Thanks!

File: app/domain/vos/account_name.go (hunk 2 above)
Comment (victoraugustolls) [reply app/domain/vos/account_name.go]: @streppel done

File: app/domain/vos/account_name.go (hunk 3 above)
Comment (victoraugustolls) [reply app/domain/vos/account_name.go]: @Streppel done!

**Review by juceliofloresta — APPROVED** [review]
:trophy: great job!

File: app/domain/vos/account_name.go (hunk 1 above)
Comment (Streppel) [reply app/domain/vos/account_name.go]: as it is structured right now I cannot think of a better name right now... I believe it'd become clearer to find a better way to represent this if we had `identifiers[0]` bound to a variable named id, like suffix, thus narrowing the scope of this slice. But this looks ok to me as it is right now

Comment (victoraugustolls) [reply app/domain/vos/account_name.go]: > I believe it'd become clearer to find a better way to represent this if we had identifiers[0] bound to a variable named id

Agree! Just changed it :)

**Review by guilhermebr — APPROVED** [review]

**Review by Streppel — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 build/
 coverage.html
 coverage.out
+.idea/
```
Comment (alexgarzao) [inline .gitignore:4]: It's specific for your tools. In this case, I think that a global git ignore is more appropriated:

https://docs.github.com/en/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer

File: app/domain/vos/account_name.go [hunk 5]
```diff
@@ -18,11 +19,16 @@ const (
...
 //   - ":" must be used as a separator.
 //
+// AccountName can have a fifth level in its structure: "class:group:subgroup:id/suffix", where:
+//   - suffix is "free text";
+//   - it is considered a suffix everything after the id
+//   - "/" must be used as a separator
+//
 // Some examples:
 //   - "assets:bacen:conta_liquidacao:tesouraria"
 //   - "liability:clients:available:96a131a8-c4ac-495e-8971-fcecdbdd003a"
```
Comment (alexgarzao) [inline app/domain/vos/account_name.go:34]: Please, put some examples with suffix here.

File: app/domain/vos/account_name.go [hunk 6]
```diff
@@ -54,18 +61,30 @@ func NewAccountName(name string) (*AccountName, error) {
 		return nil, app.ErrInvalidAccountStructure
 	}
 
+	identifiers := strings.SplitN(levels[idLevel], AccountSuffixSep, 2)
```
Comment (alexgarzao) [inline app/domain/vos/account_name.go:64]: It's ok and works well.

But what do you think about uses an auxiliary func to be more clear? Something like "ExtractIdAndSuffix" that receives the ID and split between id and suffix?

**Review by alexgarzao — APPROVED** [review]

File: app/domain/vos/account_name.go (hunk 5 above)
Comment (victoraugustolls) [reply app/domain/vos/account_name.go:34]: There are some examples bellow, should I put more?

File: app/domain/vos/account_name.go (hunk 6 above)
Comment (victoraugustolls) [reply app/domain/vos/account_name.go]: @alexgarzao done! And also added test case for empty suffix!

File: app/domain/vos/account_name.go [hunk 7]
```diff
@@ -88,3 +86,19 @@ func FormatAccount(class, group, subgroup, id, suffix string) string {
 	}
 	return name
 }
+
+func ExtractIdAndSuffix(identifier string) (string, string, error) {
+	identifiers := strings.SplitN(identifier, AccountSuffixSep, 2)
+	id := identifiers[0]
```
Comment (alexgarzao) [inline app/domain/vos/account_name.go:92]: I think that is necessary to trim spaces in id and suffix with strings.TrimSpace.

File: app/domain/vos/account_name.go [hunk 8]
```diff
@@ -61,11 +61,9 @@ func NewAccountName(name string) (*AccountName, error) {
 		return nil, app.ErrInvalidAccountStructure
 	}
 
-	identifiers := strings.SplitN(levels[idLevel], AccountSuffixSep, 2)
-	id := identifiers[0]
-	suffix := ""
-	if len(identifiers) > 1 {
-		suffix = identifiers[1]
+	id, suffix, err := ExtractIdAndSuffix(levels[idLevel])
```
Comment (alexgarzao) [inline app/domain/vos/account_name.go]: Very nice approach (clean code) :-)

File: app/domain/vos/account_name.go (hunk 5 above)
Comment (alexgarzao) [reply app/domain/vos/account_name.go:34]: Oh, my mistake.

**Comment by duarte2025** [comment]
Great work!

**Review by duarte2025 — APPROVED** [review]

File: app/domain/vos/account_name.go (hunk 8 above)
Comment (victoraugustolls) [reply app/domain/vos/account_name.go:64]: Thanks! :)

File: .gitignore (hunk 4 above)
Comment (victoraugustolls) [reply .gitignore:4]: done!

File: app/domain/vos/account_name.go (hunk 7 above)
Comment (victoraugustolls) [reply app/domain/vos/account_name.go]: Done!

File: clients/grpc/examples/get_account_balance.go [hunk 9]
```diff
@@ -71,3 +72,57 @@ func getAccountBalanceNotFoundAccount(log *logrus.Entry, conn *ledger.Connection
...
+	log.Println("starting getAccountBalanceWithWildcard")
+	defer log.Println("finishing getAccountBalanceWithWildcard")
+
+	transactionOne := 1100
+	transactionTwo := 900
+	expectedBalance := transactionOne + transactionTwo
+	newUUID := uuid.New().String()
+	accountPathOne := "liability:stone:clients:" + newUUID + "/test_1"
+	accountPathTwo := "liability:stone:clients:" + newUUID + "/test_2"
+	accountPathThree := "liability:stone:clients:" + uuid.New().String()
```
Comment (alexgarzao) [inline clients/grpc/examples/get_account_balance.go:86]: Very nice PR!

Just more one test. Try to do a transaction with "liability:stone:clients:" + newUUID, without any sufiix. In this case, the operation must be included in the balance.

**Review by alexgarzao — APPROVED** [review]

File: clients/grpc/examples/get_account_balance.go (hunk 9 above)
Comment (victoraugustolls) [reply clients/grpc/examples/get_account_balance.go]: Thanks :smile: ! And just added this test case!!

**Review by alexgarzao — APPROVED** [review]
